ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs

```diff
@@ -574,251 +559,327 @@ traceLedgerProcessing
-  --             Left err -> fail $ "benchmark doesn't support invalid headers: " <> show rp <> " " <> show err
-  --             Right x -> pure x
-
-  --       tickTheLedgerState ::
-  --            SlotNo
-  --         -> ExtLedgerState blk
-  --         -> IO (Ticked (LedgerState blk))
-  --       tickTheLedgerState slot st =
-  --           pure $ applyChainTick lcfg slot (ledgerState st)
-
-  --       applyTheBlock ::
-  --            Ticked (LedgerState blk)
-  --         -> IO (LedgerState blk)
-  --       applyTheBlock tickedLedgerSt = do
-  --           case runExcept (lrResult <$> applyBlockLedgerResult lcfg blk tickedLedgerSt) of
-  --             Left err -> fail $ "benchmark doesn't support invalid blocks: " <> show rp <> " " <> show err
-  --             Right x  -> pure x
+     HasAnalysis blk
+  => Maybe FilePath -> Analysis blk
+benchmarkLedgerOps mOutfile AnalysisEnv {db, initLedger, cfg, limit, backing, ledgerDbFS} = do
+    doCheck <- onlyCheckNumBlocks limit
+
+    initLedger' <- initialiseLedger ledgerDbFS cfg initLedger
+
+    (ldb, bs) <- ledgerDbAndBackingStore backing ledgerDbFS $ case initLedger of
+                                                                Left s -> Left (initLedger', s)
+                                                                Right v -> Right v
+
+    withFile mOutfile $ \outFileHandle -> do
+      let line = Builder.run $ DP.showHeaders separator
+                             <> separator
+                             <> "...era-specific stats"
+      Text.IO.hPutStrLn outFileHandle line
+
+      void $ runExceptT (consumeStream (mkStream doCheck GetBlock db) (castPoint . getTip $ initLedger') (ldb, 0) (process bs outFileHandle))
+      pure Nothing
+  where
+
+    lcfg = topLevelConfigLedger cfg
+    ccfg = topLevelConfigProtocol cfg
+
+    withFile :: Maybe FilePath -> (IO.Handle -> IO r) -> IO r
+    withFile (Just outfile) = IO.withFile outfile IO.WriteMode
+    withFile Nothing        = \f -> f IO.stdout
+
+    -- Separator for the data that is printed
+    separator = "\t"
+
+    process ::
+         LedgerBackingStore IO (ExtLedgerState blk)
+      -> IO.Handle
+      -> blk
+      -> (LedgerDB' blk, Word64)
+      -> IO (LedgerDB' blk, Word64)
+    process backingStore outFileHandle blk (ldb, sinceLast) = do
+        prevRtsStats <- GC.getRTSStats
+        let
+          -- Compute how many nanoseconds the mutator used from the last
+          -- recorded 'elapsedTime' till the end of the execution of the given
+          -- action. This function forces the evaluation of its argument's
+          -- result.
+          time act = do
+              tPrev <- GC.mutator_elapsed_ns <$> GC.getRTSStats
+              !r <- act
+              tNow <- GC.mutator_elapsed_ns <$> GC.getRTSStats
+              pure (r, tNow - tPrev)
+
+        let slot = blockSlot      blk
+        ((ldb', sinceLast'), tFlush) <- time $
```

Comment:
In the `OnDisk` module, it looks like the flush check is made after applying a block, instead of before it. It's probably equivalent, but the flushing time could be attributed to a slot/block that is off-by-one compared to actual flush slot/block

Reply:
I think this should be fine taken with some caution. This is just an indication of the overall performance of the system, so I don't think it is that crucial that it mimicks the behavior of the node?